pkgs/by-name/sy/syspower/package.nix

```
@@ -0,0 +1,47 @@
+{
+  stdenv,
+  lib,
+  fetchFromGitHub,
+  makeWrapper,
+  pkg-config,
+  gtkmm4,
+  gtk4-layer-shell,
+  ...
+}:
+stdenv.mkDerivation {
+  pname = "syspower";
+  version = "0-unstable-2024-12-10";
+
+  src = fetchFromGitHub {
+    owner = "System64fumo";
+    repo = "syspower";
+    rev = "b205c3443daaee6c3454f23b93b78f42c10221a1";
+    hash = "sha256-kLaWIaMTRurY7a8Nn7C9ft/Zdvx35AiU7Db8n/NFy90=";
+  };
+
+  buildInputs = [
+    gtkmm4
+    gtk4-layer-shell
+  ];
+
+  nativeBuildInputs = [
+    makeWrapper
+    pkg-config
+  ];
+
```

```suggestion

  env.NIX_LDFLAGS = "-rpath ${placeholder "out"}/lib";

```
We may be able to do this instead of wrapping LD_LIBRARY_PATH. Let me know if it works

---

What is `placeholder` here?

---

It's [`builtins.placeholder`](https://nix.dev/manual/nix/2.25/language/builtins#builtins-placeholder). It's basically `$out`, but usable anywhere -- which is good for this case since `env` attributes usually get escaped when trying to use env vars like `$out`

---

Removing the `wrapProgram` step from `installPhase` and adding the `env.NIX_LDFLAGS` line makes the program fail to start. Just to confirm, this is adding an attribute to `stdenv.mkDerivation`?

---

Yep, seems this workaround just won't work then. That's fine :)